src/common/transformations/src/transformations/common_optimizations/variadic_split_merge.cpp

```diff
@@ -0,0 +1,361 @@
+// Copyright (C) 2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/variadic_split_merge.hpp"
+
+#include <optional>
+
+#include "itt.hpp"
+#include "openvino/core/graph_util.hpp"
+#include "openvino/op/slice.hpp"
+#include "openvino/op/strided_slice.hpp"
+#include "openvino/op/variadic_split.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "transformations/utils/utils.hpp"
+
+namespace ov::pass {
+
+struct split_candidate {
+    std::shared_ptr<ov::Node> node;
+    int64_t axis;
+    int64_t begin;
+    int64_t end;
+};
+
+static bool mask_zero(const std::vector<int64_t>& mask) {
+    for (size_t i = 0; i < mask.size(); i++) {
+        if (mask[i] != 0)
+            return false;
+    }
+    return true;
+}
+
+static std::optional<int64_t> axis_scan(const std::vector<int64_t>& axis_mask) {
+    int64_t axis = 0;
+    int n = 0;
+    for (size_t i = 0; i < axis_mask.size(); i++) {
+        auto mask = axis_mask[i];
+        if (mask == 0) {
+            n++;
+            axis = i;
+        }
+        if (n >= 2)
+            return std::nullopt;
+    }
+
+    if (n == 0)
+        return std::nullopt;
+
+    return axis;
+}
+
+static std::optional<int64_t> get_scalar(const std::shared_ptr<ov::Node>& node) {
+    auto constant = ov::as_type_ptr<ov::op::v0::Constant>(node);
+    if (!constant)
+        return std::nullopt;
+    if (shape_size(constant->get_shape()) != 1)
+        return std::nullopt;
+    auto value = constant->cast_vector<int64_t>()[0];
+    return value;
+}
+
+static int64_t convert_value(int64_t value, int64_t max) {
+    if (value >= 0) {
+        if (value > max)
+            return max;
+
+        return value;
+    } else {
+        int64_t temp = max + (value + 1);
+        if (temp < 0)
+            return 0;
+
+        return max + (value + 1);
+    }
+}
+
+VariadicSplitMerge::VariadicSplitMerge() {
+    MATCHER_SCOPE(VariadicSplitMerge);
+
+    auto any_node = ov::pass::pattern::wrap_type<ov::op::Op>([](const ov::Output<ov::Node>&) {
```

Comment:
please use 
auto input_to_slice = any_input();
auto slice = wrap_type<Slice,StridedSlice>();

in the transformation callback you will be able to check all consumers of `input_to_slice` node